tests/unittests/text/test_perplexity.py

```diff
@@ -0,0 +1,79 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from functools import partial
+
+import pytest
+import torch
+import torch.nn.functional as F
+
+from torchmetrics.functional.text.perplexity import perplexity
+from torchmetrics.text.perplexity import Perplexity
+from unittests.helpers.testers import MetricTester
+from unittests.text.inputs import (
+    MASK_INDEX,
+    _logits_inputs_fp32,
+    _logits_inputs_fp32_with_mask,
+    _logits_inputs_fp64,
+    _logits_inputs_fp64_with_mask,
+)
+
+
+def _sk_perplexity(preds, target, ignore_index):
```

Comment:
the sk usually refers to sklearn so lets change it here
```suggestion
def _reference_perplexity(preds, target, ignore_index):
```